crates/core/tedge_agent/src/mqtt_operation_converter/actor.rs

```diff
@@ -0,0 +1,132 @@
+use crate::mqtt_operation_converter::error::MqttRequestConverterError;
+use async_trait::async_trait;
+use tedge_actors::fan_in_message_type;
+use tedge_actors::Actor;
+use tedge_actors::DynSender;
+use tedge_actors::LoggingReceiver;
+use tedge_actors::MessageReceiver;
+use tedge_actors::RuntimeError;
+use tedge_actors::Sender;
+use tedge_api::Jsonify;
+use tedge_api::RestartOperationRequest;
+use tedge_api::RestartOperationResponse;
+use tedge_api::SoftwareListRequest;
+use tedge_api::SoftwareListResponse;
+use tedge_api::SoftwareUpdateRequest;
+use tedge_api::SoftwareUpdateResponse;
+use tedge_mqtt_ext::MqttMessage;
+use tedge_mqtt_ext::Topic;
+
+fan_in_message_type!(AgentInput[MqttMessage, SoftwareListResponse, SoftwareUpdateResponse, RestartOperationResponse] : Debug);
+
+pub struct MqttOperationConverterActor {
+    input_receiver: LoggingReceiver<AgentInput>,
+    software_list_sender: DynSender<SoftwareListRequest>,
+    software_update_sender: DynSender<SoftwareUpdateRequest>,
+    restart_sender: DynSender<RestartOperationRequest>,
+    mqtt_publisher: DynSender<MqttMessage>,
```

Comment:
It might be better to use `LoggingSender`s.

Reply:
The file has been renamed - but the comment still applies.

Reply:
Done